otelcollector/shared/process_utilities.go

```diff
@@ -406,16 +406,18 @@ func WaitForTokenAdapter(ccpMetricsEnabled string) {
 	}
 	waitedSecsSoFar := 1
 
+	var resp *http.Response
+
 	for {
 		if waitedSecsSoFar > tokenAdapterWaitSecs {
-			if _, err := http.Get("http://localhost:9999/healthz"); err != nil {
+			if resp, err = http.Get("http://localhost:9999/healthz"); err != nil {
```

Comment:
Hi Rashmi - instead of using the global http client, can we just create a new explicit client, and when doing so, specify like a 2s timeout, so we don't piggy-back on the global default client's no timeout ?  please look for this pattern across everywhere where we are using http.<...> to make them not use the default client...

Reply:
thanks Vishwa, added. Testing them out.

Reply:
thanks.